analyzer/src/main/kotlin/YamlFilePackageCurationProvider.kt

```diff
@@ -35,5 +38,21 @@ class YamlFilePackageCurationProvider(
         curationFile.readValue<List<PackageCuration>>()
     }
 
-    override fun getCurationsFor(identifier: Identifier) = packageCurations.filter { it.id.matches(identifier) }
+    override fun getCurationsFor(packageId: Identifier): List<PackageCuration> {
+        val packageVersion = try {
+            Semver(packageId.version, Semver.SemverType.IVY)
+        } catch (e: SemverException) {
+            null
```

Comment:
Instead of swallowing this exception would it make sense to either keep it uncaught or re-throw
an uncaught exception in order to have the validity of the curation covered by the PSV?

Assuming (but haven't checked) that the IDs we currently have are all valid Semvers of type IVY.

Reply:
Why we probably currently don't have any versions in curation that are not also valid Ivy version, I'd like to not break with the current behavior where we basically allow any string as the version, so I fall back to that behavior further down if `packageVersion` is `null`.

Reply:
Can you tell why you think it's better to tolerate  any string as version?

Reply:
Because a "version" string like e.g. "2019.Jan.01" is not a valid "Ivy" (or even "Loose") Semver.

Reply:
Packages might use versions which are not valid Ivy versions, but we still need to be able to curate them.

Reply:
Now it makes sense, thanks - I misunderstood you above sentence before...